controllers/datamover.go

```diff
@@ -186,3 +203,176 @@ func (r *DPAReconciler) getDataMoverLabels() map[string]string {
+					corev1.EventTypeNormal,
+					"ResticSecretReconciled",
+					fmt.Sprintf("%s restic secret %s", op, rsecret.Name),
+				)
+			}
+		}
+		// TODO: Azure & GCP
+	}
+
+	return nil, nil
+}
+
+//build data mover restic secret for given bsl
+func (r *DPAReconciler) buildDataMoverResticSecret(rsecret *corev1.Secret, key string, secret string, region string, pass []byte, repo string) error {
+
+	// TODO: add gcp, azure support
+	rData := &corev1.Secret{
+		Data: map[string][]byte{
+			AWSAccessKey:     []byte(key),
+			AWSSecretKey:     []byte(secret),
+			AWSDefaultRegion: []byte(region),
+			ResticPassword:   pass,
+			ResticRepository: []byte(repo),
+		},
+	}
+	rsecret.Data = rData.Data
+	return nil
+}
+
+func (r *DPAReconciler) ReconcileDataMoverResticSecret(log logr.Logger) (bool, error) {
+
+	// fetch latest DPA instance
+	dpa := oadpv1alpha1.DataProtectionApplication{}
+	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
+		return false, err
+	}
+
+	// create secret only if data mover is enabled
+	if dpa.Spec.Features.DataMover != nil && dpa.Spec.Features.DataMover.Enable {
+		// obtain restic secret name from the config
+		// if no name is mentioned in the config, then assume default restic secret name
+		dmresticsecretname := ResticsecretName
+		name := dpa.Spec.Features.DataMover.CredentialName
+		if name != "" {
+			if len(name) > 0 {
+				dmresticsecretname = name
+			}
+
+		}
+
+		// fetch restic secret from the cluster
+		resticSecret := corev1.Secret{}
+		if err := r.Get(r.Context, types.NamespacedName{Namespace: r.NamespacedName.Namespace, Name: dmresticsecretname}, &resticSecret); err != nil {
+			r.Log.Error(err, "unable to fetch Restic Secret")
+			return false, err
+		}
+
+		// validate restic secret
+		val := r.validateDataMoverCredential(&resticSecret)
+		if !val {
+			return false, fmt.Errorf("no password supplied in the restic secret")
+		}
+
+		// obtain the password from user supllied restic secret
+		var res_pass []byte
+		for key, val := range resticSecret.Data {
+
+			if key == ResticPassword {
+				res_pass = val
+			}
+		}
+		// Filter bsl based on the labels and dpa name
+		// For each bsl in the list, create a restic secret
+		// Label each restic secret with bsl name
+		bslLabels := map[string]string{
+			"app.kubernetes.io/name":       common.OADPOperatorVelero,
+			"app.kubernetes.io/managed-by": common.OADPOperator,
+			"app.kubernetes.io/component":  "bsl",
+			"openshift.io/oadp":            "True",
+		}
+		bslListOptions := client.MatchingLabels(bslLabels)
+		backupStorageLocationList := velerov1.BackupStorageLocationList{}
+
+		// Fetch the configured backupstoragelocations
+		if err := r.List(r.Context, &backupStorageLocationList, bslListOptions); err != nil {
+			return false, err
+		}
+
+		for _, bsl := range backupStorageLocationList.Items {
+			if strings.Contains(bsl.Name, dpa.Name) {
+				_, err := r.createResticSecretsPerBSL(&dpa, bsl, dmresticsecretname, res_pass)
```

Comment:
@savitharaghunathan Are we assuming that the given datamover secret will work for the BSL configured ? According to this current code, if I am not wrong, we are creating a dm secret per BSL and password provided in this secret will work for all the BSLs configured ?
For e.g: 
1. OADP has 2 BSLs - AWS and GCP. User supplied one dm secret,this PR will be creating restic secret using the dm secret for both the BSLs, right ? I think we should only create the restic secret for the dm one.
2. OADP has 2 BSLs - Both AWS. Same issue.

**Suggested Solution:**
We add a BSLName and Provider field to Deature -> DataMover/ And we create the restic secret only for the BSL specified here. WDYT ?

Reply:
Please correct me if I am wrong here.